### PR TITLE
test: added route level integration testing

### PR DESCRIPTION
**Pull request by Oxiang**
## Problem

Rewrote: Route level integration testing
Migrated to this branch for new timeline

## Solution

Mocking the express app and calling the route with supertest
Mocked first layer session to bypass userguard when required
Set up mocks for each required route test

**New scripts**:

- All Route test scripts are under test/server/api

**New dev dependencies**:

- supertest
- @types/supertest

### CODE REVIEW

**Comment by Oxiang** [comment]
Rewrote commit message

File: test/server/api/LinkStatsRoute.test.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  .bind<UserRepositoryInterface>(DependencyIds.userRepository)
+  .to(MockUserRepository)
+container
+  .bind<LinkStatisticsServiceInterface>(DependencyIds.linkStatisticsService)
+  .to(LinkStatisticsServiceMock)
+
+// Importing setup app
+import app from './setup'
+
+describe('GET: /api/link-stats', () => {
```
Comment (LoneRifle) [inline test/server/api/LinkStatsRoute.test.ts:28]: nitpick - here and elsewhere:
```suggestion
describe('GET /api/link-stats', () => {
```

File: test/server/api/LinkStatsRoute.test.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  test('positive test: Should get statistics of mocked link', async (done) => {
```
Comment (LoneRifle) [inline test/server/api/LinkStatsRoute.test.ts:29]:
```suggestion
  it('should get statistics of mocked link', async (done) => {
```
The goal here is to let it read like a spec document. the messages that are passed to `it/test` are also formed as part of the test report; if you follow convention, it should look something like this in the CLI:
image

**Review by LoneRifle — CHANGES_REQUESTED** [review]
per comments

**Review by LoneRifle — APPROVED** [review]
I'm going to approve this for now, but I had hoped for a more thorough revision of the PR such that, amongst other things:
  - `test()` would have been replaced by `it()`
  - proper tense was used for the test titles

Feel free to proceed with merging when ready.

**Pull request merged**